### PR TITLE
Update changelog for v0.12.0

### PR DESCRIPTION
**Pull request by IodaMikeMapbox**
Checks:

* [x] Update changelog
* [x] Rebase to `release/0.12.0` branch
* [x] Assign reviewers

Linked issues:
mapbox/mapbox-vision#2314

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,7 +2,13 @@
 
 ## 1.0-32
 
+The latest version of the Vision Teaser App includes improvements in algorithms performance, camera calibration process and lane detection accuracy. Also the new version contains new improved ML-models for CV tasks like detection, segmentation and classification. The full list of changes:
+`MapboxVision` v0.12.0
 - Added japanese traffic signs
+- Improved preformance
```
Comment (yunikkk) [inline CHANGELOG.md:8]: typo : `preformance`

Comment (yunikkk) [reply CHANGELOG.md:8]: Also IMO sounds vary vague

Comment (IodaMikeMapbox) [reply CHANGELOG.md]: changed to `Improved performance by utilizing ARM Advanced SIMD architecture extension`, is that better wording?

Comment (yunikkk) [reply CHANGELOG.md]: Well, I would expect not implementation side of the performance improvements but user facing. So if the SIMD optimizations reduced CPU usage probably it's good to mention `Reduced CPU usage of some feature`

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,7 +2,13 @@
 
 ## 1.0-32
 
+The latest version of the Vision Teaser App includes improvements in algorithms performance, camera calibration process and lane detection accuracy. Also the new version contains new improved ML-models for CV tasks like detection, segmentation and classification. The full list of changes:
+`MapboxVision` v0.12.0
 - Added japanese traffic signs
+- Improved performance by utilizing ARM Advanced SIMD architecture extension
+- Improved lanes detection
+- Improved camera calibration algorithm
+- Released new models version v4
```
Comment (yunikkk) [inline CHANGELOG.md:11]: Seems user does not know about versions of models - suggest to rephrase it. Maybe
`Utilized new ML models that reduce resource utilization`

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,7 +2,13 @@
 
 ## 1.0-32
 
+The latest version of the Vision Teaser App includes improvements in algorithms performance, camera calibration process and lane detection accuracy. Also the new version contains new improved ML-models for CV tasks like detection, segmentation and classification. The full list of changes:
+`MapboxVision` v0.12.0
 - Added japanese traffic signs
+- Improved performance by utilizing ARM Advanced SIMD architecture extension
+- Improved lanes detection
```
Comment (dersim-davaod) [inline CHANGELOG.md:9]: nit:
```suggestion
- Improved lanes detection algorithm
```

File: CHANGELOG.md (hunk 1 above)
Comment (chezzdev) [reply CHANGELOG.md]: Wait, this is release notes for teaser, who cares about SIMD and Neon? I think "Overall performance improvements" are good enough. Or if it seems vague, we can remove it completely.

Comment (kiryldz) [reply CHANGELOG.md]: I'd remove it completely (even `Performance improvements`). Definitely mentioning SIMD, Neon etc is not needed. I've just speeded up existing Neon a bit, it's not noticeable at all from user side.

Comment (IodaMikeMapbox) [reply CHANGELOG.md]: ok, removed

File: CHANGELOG.md [hunk 4]
```diff
@@ -2,7 +2,12 @@
 
 ## 1.0-32
 
+The latest version of the Vision Teaser App includes improvements in algorithms performance, camera calibration process and lane detection accuracy. Also the new version contains new improved ML-models for CV tasks like detection, segmentation and classification. The full list of changes:
```
Comment (chezzdev) [inline CHANGELOG.md:5]:
```suggestion
The latest version of the Vision Teaser includes improvements in algorithm performance, camera calibration process, and lane detection accuracy. The new version contains new improved ML-models for CV tasks like detection, segmentation, and classification. The full list of changes:
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -2,7 +2,12 @@
...
+- `MapboxVision` v0.12.0
 - Added japanese traffic signs
```
Comment (chezzdev) [inline CHANGELOG.md:7]:
```suggestion
- Added detection of Japanese traffic signs
```

File: CHANGELOG.md [hunk 6]
```diff
@@ -2,7 +2,12 @@
...
+- Improved lanes detection algorithm
+- Improved camera calibration algorithm
+- Utilized new ML models that reduce resource utilization
```
Comment (chezzdev) [inline CHANGELOG.md:10]:
```suggestion
- Utilized new ML models that reduce resource consumption
```
To remove "utilize" repetition.

**Pull request merged**